packages/2d/src/components/SVG.ts

```diff
@@ -0,0 +1,387 @@
+
+    if (bIndex !== bNodes.length) diff.inserted.push(...bNodes.slice(bIndex));
+
+    diff.deleted = diff.deleted.filter(aNode => {
+      const bIndex = diff.inserted.findIndex(bNode =>
+        this.isNodeEqual(aNode, bNode),
+      );
+      if (bIndex >= 0) {
+        const bNode = diff.inserted[bIndex];
+        diff.inserted.splice(bIndex, 1);
+        diff.transformed.push({
+          from: aNode,
+          to: bNode,
+        });
+
+        return false;
+      }
+
+      return true;
+    });
+    return diff;
+  }
+
+  private cloneNodeExact(node: Node) {
+    const props: ShapeProps = {
+      position: node.position(),
+      scale: node.scale(),
+      rotation: node.rotation(),
+    };
+    if (node instanceof Layout) {
+      props.size = node.size();
+    }
+    return node.clone(props);
+  }
+
+  public *tweenSVG(svg: string, time: number) {
+    const newSVG = this.parseSVG(svg);
+    const diff = this.diffSVG(this.parsed(), newSVG);
+    const transformed = diff.transformed.map(({from, to}) => ({
+      from: this.cloneNodeExact(from),
+      current: from,
+      to,
+    }));
+
+    for (const node of diff.inserted) this.wrapper.add(node);
+
+    const autoWidth = this.customWidth() == null;
+    const autoHeight = this.customHeight() == null;
+
+    const beginning = 0.2;
+    const ending = 0.8;
+    const overlap = 0.15;
+
+    yield* tween(
+      time,
+      value => {
+        const remapped = clampRemap(beginning, ending, 0, 1, value);
+        const eased = easeInOutSine(remapped);
+        for (const node of transformed) {
+          node.current.position(
+            Vector2.lerp(node.from.position(), node.to.position(), eased),
+          );
+          node.current.scale(
+            Vector2.lerp(node.from.scale(), node.to.scale(), eased),
+          );
+          node.current.rotation(
+            easeInOutSine(remapped, node.from.rotation(), node.to.rotation()),
+          );
+
+          if (node.current instanceof Rect) {
+            node.current.size(
+              Vector2.lerp(
+                (node.from as Layout).size(),
+                (node.to as Layout).size(),
+                eased,
+              ),
+            );
+          }
+        }
+
+        const scale = this.wrapper.scale();
+        if (autoWidth)
+          this.customWidth(
+            easeInOutSine(value, diff.fromSize.x, diff.toSize.x) * scale.x,
+          );
+
+        if (autoHeight)
+          this.customHeight(
+            easeInOutSine(value, diff.fromSize.y, diff.toSize.y) * scale.y,
+          );
+
+        const deletedOpacity = clampRemap(0, beginning + overlap, 1, 0, value);
+        for (const node of diff.deleted) node.opacity(deletedOpacity);
+
+        const insertedOpacity = clampRemap(ending - overlap, 1, 0, 1, value);
+        for (const node of diff.inserted) node.opacity(insertedOpacity);
+      },
+      () => {
+        this.wrapper.children(this.parsedNodes);
```

Comment:
Ideally, we should also call `dispose` on the removed nodes and the clones used for transformed nodes